cnd/src/db/save.rs

```diff
@@ -505,3 +505,28 @@ impl Save<Decline> for Sqlite {
         Ok(())
     }
 }
+
+#[async_trait]
+impl<A, B> Save<CreatedSwap<A, B>> for Sqlite
+where
+    A: Save<A>,
```

Comment:
This is an interesting approach, I like it.

I do think though that this is optimizing too early. For now, I would just write only a `Save` impl where `A` and `B` are filled in.
Doing that will ultimately lead to quite a bit of code duplication between those `Save` impls. I suspect that we should be able to create another trait that goes on these bounds that allows us to write a single `Save` impl where `A` and `B` fulfill this trait. I am pretty certain that that one would be internal to the database though and not the `Save` trait but more something like a conversion to some internal DB model and then calling to diesel.